src/main/java/dal/fatih/todoly/Main.java

```diff
@@ -55,28 +55,54 @@ public static void listAllTasks() {
             }
         }
     }
+
     public static void showTaskDetails() {
         System.out.print("Task Id :");
         String taskId = scn.nextLine();
-        Task task=tasks.get(taskId);
+        Task task = tasks.get(taskId);
         if (task != null) {
             System.out.println(tasks.get(taskId));
         } else {
             System.out.println("Task not found");
         }
     }
-    public static void deleteTask(){
+
+    public static void deleteTask() {
         System.out.print("Task Id :");
         String taskId = scn.nextLine();
-        Task task=tasks.get(taskId);
+        Task task = tasks.get(taskId);
         if (task != null) {
             tasks.remove(taskId);
-            System.out.println(task.getTitle() +"titled task deleted");
+            System.out.println(task.getTitle() + "titled task deleted");
         } else {
             System.out.println("Task not found");
         }
     }
-    
+
+    public static void filterTask() {
+
+        try {
+            DateFormat dueDateParser = new SimpleDateFormat("dd/MM/yyyy");

```

Comment:
Create only one parser and keep using it for other options too. Don't recreate it all the times